cloudinit/sources/helpers/vultr.py

```diff
@@ -0,0 +1,293 @@
+# Author: Eric Benner <ebenner@vultr.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import copy
+import base64
+
+from cloudinit import log as log
+from cloudinit import url_helper
+from cloudinit import dmi
+from cloudinit import util
+from cloudinit import net
+from cloudinit.net.dhcp import EphemeralDHCPv4, NoDHCPLeaseError
+from functools import lru_cache
+
+# Get LOG
+LOG = log.getLogger(__name__)
+
+
+@lru_cache()
+def get_metadata(params):
```

Comment:
I ended up having to do this because lr_cache throws not hashable errors when a dict is either passed or returned. Figured this was a little more readable then the solutions I was seeing about getting around it.

https://stackoverflow.com/questions/6358481/using-functools-lru-cache-with-dictionary-arguments

Up to you on how you want me to go about this.

Reply:
This may not matter in light of other changes.